src/java_bytecode/java_types.cpp

```diff
@@ -246,11 +246,7 @@ typet java_type_from_string(const std::string &src)
           class_name[i]='.';
 
       std::string identifier="java::"+class_name;
-
-      reference_typet result;
-      result.subtype()=symbol_typet(identifier);
-      result.subtype().set(ID_C_base_name, class_name);
-      return result;
+      return java_reference_type(symbol_typet(identifier));
```

Comment:
Here we fail to set `ID_C_base_name` on the symbol which somehow results in Java types not being added to the symbol table. Changing to:

```C++
symbol_typet new_symbol=symbol_typet(identifier);
new_symbol.set(ID_C_base_name, class_name);
return java_reference_type(new_symbol);
```

Reply:
I'd say that needs a cleanup, but that should become a different PR.